### PR TITLE
[Remove Stream/Optional usage] LPS-178007 remove_stream_and_optional_in_headless_delivery_api

### PR DESCRIPTION
**Pull request by jeffwu0724**
Hi Tina, Dante,

Fixed the code conflict and resent: https://github.com/liferay-core-infra/liferay-portal/pull/3814
This is for ticket https://issues.liferay.com/browse/LPS-178007.
Thanks for checking!

### CODE REVIEW

**Comment by jeffwu0724** [comment]
ci:test:headless

**Comment by catarineaguiar** [comment]
no regressions found.

File: modules/apps/headless/headless-delivery/headless-delivery-api/src/main/java/com/liferay/headless/delivery/dto/v1_0/util/ContentFieldUtil.java [hunk 1]
```diff
@@ -114,14 +114,13 @@ public static ContentField toContentField(
...
+						Value value = ddmFormFieldValue.getValue();
 
-						Map<Locale, String> valueValues = Optional.ofNullable(
-							ddmFormFieldValue.getValue()
-						).map(
-							Value::getValues
-						).orElse(
-							Collections.emptyMap()
-						);
+						Map<Locale, String> valueValues = new HashMap<>();
```
Comment (dantewang) [inline modules/apps/headless/headless-delivery/headless-delivery-api/src/main/java/com/liferay/headless/delivery/dto/v1_0/util/ContentFieldUtil.java:119]: @jeffwu0724 Originally it returns `Collections.emtpyMap()` as fallback, please be consistent.

Also, please try to refine the logic and naming here.

File: modules/apps/headless/headless-delivery/headless-delivery-api/src/main/java/com/liferay/headless/delivery/dto/v1_0/util/DDMValueUtil.java [hunk 2]
```diff
@@ -448,23 +446,21 @@ private static LocalizedValue _toLocalizedValue(
 			preferredLocale,
 			localizedValueBiFunction.apply(contentFieldValue, preferredLocale));
 
-		Optional.ofNullable(
-			localizedContentFieldValues
-		).orElse(
-			Collections.emptyMap()
-		).forEach(
-			(languageId, localizedContentFieldValue) -> {
+		if (localizedContentFieldValues != null) {
```
Comment (dantewang) [inline modules/apps/headless/headless-delivery/headless-delivery-api/src/main/java/com/liferay/headless/delivery/dto/v1_0/util/DDMValueUtil.java:449]: @jeffwu0724 Please flip.

**Review by dantewang — CHANGES_REQUESTED** [review]
@jeffwu0724 Please see my comment and resend, please make sure you do a full scan to review your changes before you resend, thanks!

**Pull request closed without merging**